OpenRA.Mods.Common/Traits/World/AutoSave.cs

```diff
@@ -0,0 +1,94 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[TraitLocation(SystemActors.World)]
+	[Desc("Add this trait to the world actor to enable auto-save.")]
+	public class AutoSaveInfo : TraitInfo
+	{
+		public override object Create(ActorInitializer init) { return new AutoSave(init.Self, this); }
+	}
+
+	public class AutoSave : ITick
+	{
+		const string AutoSavePattern = "auto-save@";
+		const string SaveFileExtension = ".orasav";
+		int ticksUntilAutoSave;
+		int lastSaveInverval;
+
+		public AutoSave(Actor self, AutoSaveInfo info)
+		{
+			ticksUntilAutoSave = GetTicksBetweenAutosaves(self);
+			lastSaveInverval = Game.Settings.SinglePlayerSettings.AutoSaveInterval;
+		}
+
+		void ITick.Tick(Actor self)
+		{
+			var autoSaveValue = Game.Settings.SinglePlayerSettings.AutoSaveInterval;
+
+			if (autoSaveValue == 0) return;
```

Comment:
```suggestion
			if (autoSaveValue == 0) 
					return;
```

this is our style convention